utils/table_printer.go

```diff
@@ -23,227 +21,71 @@ type TablePrinterOptions struct {
 	Out      io.Writer
 }
 
+// Deprecated: use internal/tableprinter
 func NewTablePrinter(io *iostreams.IOStreams) TablePrinter {
 	return NewTablePrinterWithOptions(io, TablePrinterOptions{
 		IsTTY: io.IsStdoutTTY(),
 	})
 }
 
+// Deprecated: use internal/tableprinter
 func NewTablePrinterWithOptions(ios *iostreams.IOStreams, opts TablePrinterOptions) TablePrinter {
 	var out io.Writer
 	if opts.Out != nil {
 		out = opts.Out
 	} else {
 		out = ios.Out
 	}
+	var maxWidth int
 	if opts.IsTTY {
-		var maxWidth int
 		if opts.MaxWidth > 0 {
 			maxWidth = opts.MaxWidth
 		} else if ios.IsStdoutTTY() {
 			maxWidth = ios.TerminalWidth()
 		} else {
 			maxWidth = ios.ProcessTerminalWidth()
 		}
-		return &ttyTablePrinter{
-			out:      out,
-			maxWidth: maxWidth,
-		}
 	}
-	return &tsvTablePrinter{
-		out: out,
+	tp := tableprinter.New(out, opts.IsTTY, maxWidth)
+	return &printer{
+		tp:    tp,
+		isTTY: opts.IsTTY,
 	}
 }
 
-type tableField struct {
-	Text         string
-	TruncateFunc func(int, string) string
-	ColorFunc    func(string) string
-}
-
-func (f *tableField) DisplayWidth() int {
-	return text.DisplayWidth(f.Text)
+type printer struct {
+	tp       tableprinter.TablePrinter
+	colIndex int
+	isTTY    bool
 }
 
-type ttyTablePrinter struct {
-	out      io.Writer
-	maxWidth int
-	rows     [][]tableField
+func (p printer) IsTTY() bool {
+	return p.isTTY
 }
 
-func (t ttyTablePrinter) IsTTY() bool {
-	return true
-}
-
-func (t *ttyTablePrinter) AddField(s string, truncateFunc func(int, string) string, colorFunc func(string) string) {
+func (p *printer) AddField(s string, truncateFunc func(int, string) string, colorFunc func(string) string) {
```

Comment:
Does this method belong on the internal `tableprinter`? In its current location, if I want to create a new table using the internal `tableprinter` and use `AddField` I don't get the same logic as before with the preserving of the first column and URLs. Unless you are thinking going forward we are going to remove that logic? 

I think the same needs to happen for `EndRow()` method as well since it contains logic dealing with current column index.

Reply:
This AddField is here purely for backwards compatibility with implicit behavior around avoiding truncating the first column or URL columns. Going forward, new code should import `internal/tableprinter` and explicitly mark values that it doesn't wish truncated. I did not want this behavior to be implicit (and therefore "hidden") anymore.